galaxy_ng/app/api/v3/viewsets/namespace.py

```diff
@@ -58,6 +59,12 @@ def get_serializer_class(self):
     @transaction.atomic
     def create(self, request, *args, **kwargs):
         """Override to also create inbound pulp repository and distribution."""
+
+        if models.Namespace.objects.filter(name=request.data['name']).exists():
+            raise ValidationError(detail={
+                'name': 'There is an existing namespace with the same name, choose a different name'
+            })
```

Comment:
This validation has to be done here because this `create` method is called before the validators defined on the serializer.

another option if we want to keep validation on the serializer will be moving the creation of Repo and Distro to 
there so there will be validation happening in a single place.

Reply:
IMO here 409 conflict response code is more appropriate.

Reply:
@rochacbruno 

> another option if we want to keep validation on the serializer will be moving the creation of Repo and Distro to there so there will be validation happening in a single place.

Yeah, possibly. That was debated originally iirc. I think the overall thinking was to try to keep db stuff out of the serializers as far as possible (and in the viewsets). Another common option seems to be to move the "create three different model instances at the same time" stuff into one (or more) of the model objectManagers. 

So this seems reasonable for now, with a note to revisit it.

Reply:
Would this be a place to apply ScopedErrorListSerializer https://github.com/ansible/galaxy_ng/blob/master/galaxy_ng/app/api/v3/serializers/namespace.py#L17 ?

Reply:
@cutwater If 409 is more appropriate here (and I agree that it is), it feels like letting this hit a db error and catching the exception somewhere appropriate would be my expectation.

Reply:
Resolved as a 409 conflict error

Reply:
@alikins changed to use try/except as you suggested so it goes EAFP and hit the database